plaso/parsers/dsv_parser.py

```diff
@@ -134,35 +203,28 @@ def ParseFileObject(self, parser_mediator, file_object):
     Raises:
       UnableToParseFile: when the file cannot be parsed.
     """
-    file_size = file_object.get_size()
-    # The csv module can consume a lot of memory, 1 GiB for a 100 MiB file.
-    # Hence that the maximum supported file size is restricted.
-    if file_size > self._MAXIMUM_SUPPORTED_FILE_SIZE:
-      display_name = parser_mediator.GetDisplayName()
-      raise errors.UnableToParseFile((
-          '[{0:s}] Unable to parse DSV file: {1:s} size of file exceeds '
-          'maximum supported size').format(self.NAME, display_name))
-
     # TODO: Replace this with detection of the file encoding via byte-order
     # marks. Also see: https://github.com/log2timeline/plaso/issues/1971
     if not self._encoding:
       self._encoding = parser_mediator.codepage
 
-    # The Python 2 csv module reads bytes and the Python 3 csv module Unicode
-    # reads strings.
-    if py2to3.PY_3:
-      line_reader = text_file.TextFile(file_object, encoding=self._encoding)
-    else:
-      line_reader = line_reader_file.BinaryLineReader(file_object)
-
-    # If we specifically define a number of lines we should skip, do that here.
-    for _ in range(0, self.NUMBER_OF_HEADER_LINES):
-      line_reader.readline()
-
-    reader = self._CreateDictReader(line_reader)
+    try:
+      if not self._HasExpectedLineLength(file_object):
+        display_name = parser_mediator.GetDisplayName()
+        raise errors.UnableToParseFile(
+            '[{0:s}] Unable to parse DSV file: {1:s} with error: '
+            'unexpected line length.'.format(
```

Comment:
line 216 and 217 looks like they can fit on a single line.

Reply:
Also please put a multi line string in between ( .. )

Reply:
Done